controllers/app.controllers.js

```diff
@@ -1,55 +1,128 @@
-        const topics = await getAllTopics();
-        res.status(200).send({topics});
-    } catch(err) {
-        next(err);
-    }
-}
+  try {
+    const topics = await fetchAllTopics();
+    res.status(200).send({ topics });
+  } catch (err) {
+    next(err);
+  }
+};
 
 exports.getUsernames = async (req, res, next) => {
-    try {
-        const usernames = await getAllUsernames();
-        res.status(200).send({usernames});
-    } catch(err) {
-        next(err);
-    }
-}
-
+  try {
+    const usernames = await fetchAllUsernames();
+    res.status(200).send({ usernames });
+  } catch (err) {
+    next(err);
+  }
+};
 
 exports.getArticles = async (req, res, next) => {
     try {
-        const articles = await getAllArticles();
-        res.status(200).send({articles});
-    } catch {
-        next(err);
-    }
-}
+        
+    const { sort_by } = req.query;
+    const { order } = req.query;
+    const { topic } = req.query;
+    
+      if (order) {
+        const validOrders = ['ASC', 'ASCENDING', 'DESC', 'DESCENDING'];
+        if(!validOrders.includes(order) )
+          return next({status: 400, msg: 'Bad Request'})
+        }
+      
 
+    const articles = await fetchAllArticles(sort_by, order, topic);
+    res.status(200).send({ articles });
+  } catch(err) {
+    next(err);
+  }
+};
 
 exports.getArticleById = async (req, res, next) => {
-    const { article_id } = req.params;
-    try {
-        const article = await getArticleFromDbById(article_id);
-        res.status(200).send({article});
-    } catch(err) {
-        next(err);
-    }
-}
+  const { article_id } = req.params;
+  try {
+    const article = await fetchArticleFromDbById(article_id);
+    res.status(200).send({ article });
+  } catch (err) {
+    next(err);
+  }
+};
 
 exports.getCommentsForArticleById = async (req, res, next) => {
-    const { article_id } = req.params;
-    try {
-        const article =  await getArticleFromDbById(article_id);
-        if(article) {
-            const comments = await getCommentsForArticle(article_id);
-                res.status(200).send({comments}); 
-        }
-    } catch(err) {
-        next(err);
+  const { article_id } = req.params;
+  try {
+    const article = await fetchArticleFromDbById(article_id);
+    if (article) {
+      const comments = await fetchCommentsForArticle(article_id);
+      res.status(200).send({ comments });
+    }
+  } catch (err) {
+    next(err);
+  }
+};
+
+exports.postCommentByArticleId = async (req, res, next) => {
+  if (
+    !req.body.name ||
+    !req.body.body ||
+    typeof req.body.name !== 'string' ||
+    typeof req.body.body !== 'string'
+  ) {
+    return next({ status: 400, msg: 'Bad Request' });
+  }
+
+  const { name, body } = req.body;
+  const { article_id } = req.params;
+
+  const checkArticleExists = fetchArticleFromDbById(article_id);
+  const checkUser = checkUserExistsByName(name);
+
+  return Promise.all([checkArticleExists, checkUser, article_id, name, body])
```

Comment:
Some lovely work here but the database will protect you from inserting missing values here as they are FOREIGN KEYs. As such you can't insert values that don't exists in the other table. 

No need to change what you've done here though. It's still good work.